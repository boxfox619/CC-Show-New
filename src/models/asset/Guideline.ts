export class Guideline {
    constructor(
        public x: number,
        public y: number,
        public width: number,
        public height: number
    ) { }

    get style() {
        return {
            left: this.x + 'px',
            top: this.y + 'px',
            width: this.width > 0 ? this.width + 'px' : undefined,
            height: this.height > 0 ? this.height + 'px' : undefined,
        }
    }
}

export default Guideline;