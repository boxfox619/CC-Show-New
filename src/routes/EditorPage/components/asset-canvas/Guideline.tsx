import * as React from 'react';
import GuidelineModel from '../../models/Guideline';
import styled from 'styled-components';
import { THEME_COLOR } from '../../../../util/constant';

const Line = styled.div`
    position: absolute;
    ${(props: { vertical: boolean }) => props.vertical ? `
        height: 2px;
        width: 2px;
        border-left: dashed ${THEME_COLOR} 2px;
    ` : `
        width: 2px;
        height: 2px;
        border-top: dashed ${THEME_COLOR} 2px;
    `}
`

interface Props {
    attr: GuidelineModel
}

export const Guideline: React.FC<Props> = (props) => {
    const vertical = (!!props.attr.height);
    return (<Line vertical={vertical} style={props.attr.style} />);
}