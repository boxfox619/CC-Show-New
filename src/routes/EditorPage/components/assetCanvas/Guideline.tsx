import * as React from 'react';
import GuidelineModel from '../../models/Guideline';
import styled from 'styled-components';

const Line = styled.div`
    position: absolute;
    ${(props: { vertical: boolean }) => props.vertical ? `
        height: 2px;
        width: 2px;
        border-left: dashed #038AFD 2px;
    ` : `
        width: 2px;
        height: 2px;
        border-top: dashed #038AFD 2px;
    `}
`

interface Props {
    attr: GuidelineModel
}

export const Guideline: React.FC<Props> = (props: Props) => {
    const vertical = (!!props.attr.height);
    return (
        <Line vertical={vertical} style={props.attr.style}/>
    )
}