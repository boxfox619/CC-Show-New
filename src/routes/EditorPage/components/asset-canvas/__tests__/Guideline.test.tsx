import * as React from 'react';
import * as Enzyme from 'enzyme';
import GuidelineModel from '../../../models/Guideline';
import { Guideline } from '../Guideline';

describe('<Guideline />', () => {
    it('should match snapshot', () => {
        const guideline = new GuidelineModel(0, 0, 0, 0);
        const wrapper = Enzyme.shallow(<Guideline key={"test"} attr={guideline}/>);
        expect(wrapper).toMatchSnapshot();
    });
});