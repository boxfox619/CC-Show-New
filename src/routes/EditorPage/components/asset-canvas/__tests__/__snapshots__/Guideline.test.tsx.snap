// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Guideline /> should match snapshot 1`] = `ShallowWrapper {}`;
